src/api/search.api.js

```diff
@@ -272,6 +279,137 @@ async function searchLayerFeatures(outputProjection, queryString, layer, lang, c
+        .toLowerCase()
+        .normalize('NFD')
+        .replace(/[\u0300-\u036f]/g, '')
+    return searchFields.some((field) => {
+        const value = feature.values_[field]?.toString()
+        return !!value && value.trim().toLowerCase().includes(queryStringClean)
+    })
+}
+
+/**
+ * Searches for the query string in the KML layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchKmlLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseKml(layer, outputProjection, [])
+        if (!features || !features.length) return []
+
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) return []
+
+        const extent = getKmlExtent(layer.kmlData)
+
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, extent, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Searches for the query string in the GPX layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchGpxLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseGpx(layer.gpxData, outputProjection, [])
+        if (!features || !features.length) return []
+
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) return []
+
+        const extent = getGpxExtent(layer.gpxData)
+
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, extent, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Creates the SearchResult for a layer
+ *
+ * @param layer
+ * @param feature
+ * @param {ol/extent|null} extent
+ * @param {CoordinateSystem} outputProjection
+ * @returns {SearchResult}
+ */
+function createSearchResultFromLayer(layer, feature, extent, outputProjection) {
+    const featureName = feature.values_.name || layer.name || ''
+    const coordinates = extractOlFeatureCoordinates(feature)
+    const zoom = outputProjection.get1_25000ZoomLevel()
+    // TODO is this the correct way we should handle this? It works fine for polygons but for lines it might be a bit off
```

Comment:
Is this the correct way of handling finding the point on which the search result should zoom to ?

Reply:
I would say : if there's an "valid" extent (meaning it has different bottomLeft/topRight coordinates) you could run a `zoomToExtent` action on that. It is otherwise a good choice to go with the current projection\s 1:25'000 zoom level 👍